src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenUsingDeclarationTests.cs

```diff
@@ -658,5 +975,118 @@ .maxstack  1
   IL_003b:  ret
 }");
         }
+
+        [Fact]
+        public void JumpBackOverUsingDeclaration()
+        {
+            string source = @"
+using System;
+class C1 : IDisposable
+{
+    private string name;
+    public C1(string name)
+    {
+        this.name = name;
+    }
+    public void Dispose()
+    {
+        Console.WriteLine(""Disposed "" + name);
+    }
+}
+class C2
+{
+    public static void Main()                                                                                                           
+    {
+        int x = 0;
+        label1:
+        {
+            using C1 o1 = new C1(""first"");
+            if(x++ < 3)
+            {
+                goto label1;
+            }
+        }
+    }
+}";
+            var output = @"Disposed first
+Disposed first
+Disposed first
+Disposed first";
+            CompileAndVerify(source, expectedOutput: output).VerifyIL("C2.Main", @"
+{
+  // Code size       36 (0x24)
+  .maxstack  3
+  .locals init (int V_0, //x
+                C1 V_1) //o1
+  IL_0000:  ldc.i4.0
+  IL_0001:  stloc.0
+  IL_0002:  ldstr      ""first""
+  IL_0007:  newobj     ""C1..ctor(string)""
+  IL_000c:  stloc.1
+  .try
+  {
+    IL_000d:  ldloc.0
+    IL_000e:  dup
+    IL_000f:  ldc.i4.1
+    IL_0010:  add
+    IL_0011:  stloc.0
+    IL_0012:  ldc.i4.3
+    IL_0013:  bge.s      IL_0017
+    IL_0015:  leave.s    IL_0002
+    IL_0017:  leave.s    IL_0023
+  }
+  finally
+  {
+    IL_0019:  ldloc.1
+    IL_001a:  brfalse.s  IL_0022
+    IL_001c:  ldloc.1
+    IL_001d:  callvirt   ""void System.IDisposable.Dispose()""
+    IL_0022:  endfinally
+  }
+  IL_0023:  ret
+}
+");
+        }
+
+        [Fact]
+        public void UsingVariableDiscardIsStillDisposed()
+        {
+            var source = @"
+using System;
+class C1 : IDisposable
+{
+    public void Dispose() { Console.Write(""Disposed""); }
+}
+class C2
+{
+    public static void Main()
+    {
+        using var _ = new C1();
```

Comment:
>var _ [](start = 14, length = 5)

Is this is a discard or a local named "_"?

Reply:
Removed. This was a sleepy me being stupid last night and just testing a variable named discard.